### PR TITLE
Document version locking mechanism

### LINKED ISSUE #40379: Add documentation regarding version lock
## Details

In v15.1.10+, a version locking mechanism was introduced into the Teleport updater. This was done as a short-term workaround to prevent unintentional updates of the Teleport agent to an incompatible version.

There was an incident where a user updated their Teleport updater to v15.1.10+, which then enabled the version lock. But they were then unable to downgrade their Teleport agent and were confused as to why or how to bypass the version lock.

## How will we know this is resolved?

The documentation should include a section about the new version locking mechanism. The section should include instructions about how to bypass the version lock for the different package managers.

### PR DESCRIPTION
**Pull request by bernardjkim**
Closes https://github.com/gravitational/teleport/issues/40379, https://github.com/gravitational/teleport/issues/36815.

This PR updates the documentation for automatic updates. A new section is added to explain the new version locking mechanism introduced in Teleport 15.1.10+. Instructions on how to bypass the lock are also included.

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

**Review by ptgott — COMMENTED** [review]
Instead of updating the architecture page, would it make sense to incorporate the `dry-run`, version locking, and updater updating information in the how-to guides for updating (pages in https://goteleport.com/docs/upgrading/)? That way, users looking for information on operating the updater (rather than conceptual information on how the updater works) don't miss this information.

**Review by fspmarshall — APPROVED** [review]

**Review by fheinecke — APPROVED** [review]

**Comment by ptgott** [comment]
Hi @bernardjkim, this has received a few approvals already, so please go ahead and merge without a second review from me!

File: docs/pages/upgrading/cloud-linux.mdx
Comment (ptgott) [inline docs/pages/upgrading/cloud-linux.mdx:167]: Currently, the default installation flow shown in Step 2 does not include version locking. I would make version locking the default path and:

- Illustrate version locking when we tell the user to install `teleport-ent` earlier in this guide (the Tabs in Step 2).
- Add a brief sentence next to the installation command snippet indicating that we recommend version locking to prevent unintended updates.

We can then add a `Details` box below the installation instructions with the content of the `Bypass version lock` section, since these are intended for a more niche subset of the target audience.

In general, I think it would make sense to integrate the version locking instructions into the rest of the guide, rather than organize it to appear as a separate alternative.

File: docs/pages/upgrading/cloud-linux.mdx
Comment (ptgott) [inline docs/pages/upgrading/cloud-linux.mdx:170]: > Due to the fact that Teleport has strict [version compatibility requirements](../faq.mdx#which-version-of-teleport-is-supported),
the updater requires full control over the version of Teleport.

Reading the compatibility requirements is already a prerequisite of the guide, so I think we can remove this sentence.

File: docs/pages/upgrading/cloud-linux.mdx
Comment (bernardjkim) [reply docs/pages/upgrading/cloud-linux.mdx:167]: The version lock is automatically enabled when the teleport-ent-updater v15.1.10+ is installed or updated. Action is not required by the user to enable the version lock.

For most users, this version lock should not change their workflow. But there can be some instances when a user may want to manually update the teleport-ent package, which would then require these extra flags/steps required.

Comment (bernardjkim) [reply docs/pages/upgrading/cloud-linux.mdx:167]: How about I put this under a FAQ section

Comment (ptgott) [reply docs/pages/upgrading/cloud-linux.mdx]: I think my comment may have came from a misconception about version locking. Is the behavior with version locking that:
- You specify a version of the Teleport package when installing via a package manager (as the original guide shows in the package manager instructions in Step 2)
- The updater locks the version specified and prevents manual updates

If that's the case, I think we can remove the "FAQ" heading and elevate "Version locking" and "Updating the updater" to H2s, then merge the PR. I don't think we need the "FAQ" heading since the headings below it make sense as H2s in their own right.

Comment (bernardjkim) [reply docs/pages/upgrading/cloud-linux.mdx]: Yup, that's the expected behavior. We're just trying to prevent any unintentional updates of the teleport agent after it has been installed. The updater will lock the version and handle all future updates.

File: docs/pages/upgrading/cloud-linux.mdx
Comment (zmb3) [inline docs/pages/upgrading/cloud-linux.mdx:169]:
```suggestion
As of Teleport `15.1.10`, a version locking mechanism is enabled by the updater.
```

I would either say "as of 15.1.10" or "Teleport versions 15.1.10+"

File: docs/pages/upgrading/cloud-linux.mdx
Comment (zmb3) [inline docs/pages/upgrading/cloud-linux.mdx:192]:
```suggestion
With the Zypper package manager CLI, the lock must be disabled and then re-enabled
```

**Review by zmb3 — APPROVED** [review]

**Review by ptgott — APPROVED** [review]

**Pull request merged**
